indico/web/forms/fields/principals.py

```diff
@@ -30,16 +32,19 @@
 
 
 def serialize_principal(principal):
+    from indico.modules.categories.util import serialize_category_role
     if principal.principal_type == PrincipalType.email:
         return serialize_email_principal(principal)
     elif principal.principal_type == PrincipalType.network:
         return serialize_ip_network_group(principal)
     elif principal.principal_type == PrincipalType.user:
         return serialize_user(principal)
     elif principal.principal_type == PrincipalType.event_role:
-        return serialize_role(principal)
+        return serialize_event_role(principal)
     elif principal.is_group:
         return serialize_group(principal)
+    elif principal.principal_type == PrincipalType.category_role:
+        return serialize_category_role(principal)
```

Comment:
I'd move this up two lines to have category roles right after event roles (it's a bit strange to have groups in between).